### DISCUSSION TITLE
How does "extract words" work?

### DISCUSSION THREAD

**Discussion by mangu75** [jsvine/pdfplumber#412] [Ask for help with specific PDFs]
Hi, 
i´m using extract word in order to get all the words from a pdf but i´m not sure how exactly it works.

This is the way i´m using it:
pdfplumber.utils.extract_words(pdf_content.chars, x_tolerance=1, y_tolerance=1, keep_blank_chars=True)

I understand that it saves in a list every group of words separated just by one space, however sometimes it appends words that have more than one space away such as this one:

![image](https://user-images.githubusercontent.com/58860850/114419151-79cf1d80-9bb3-11eb-9581-3ba83e88ee37.png)

![image](https://user-images.githubusercontent.com/58860850/114419362-a4b97180-9bb3-11eb-8812-eac5903a4fcc.png)

what is the explanation?

Thank you!!

**Comment by jsvine** [comment]
Hello, and apologies for the late response — I lost track of this inquiry. It's a bit difficult to say with certainty without having access to the PDF, but I believe the reason you are seeing words with multiple spaces is that you have set `keep_blank_chars=True`. There are two types of things that *look like* whitespace in PDFs: (1) literal blank characters, and (2) the *absence* of characters.

By default, `extract_words(...)` uses both (1) and (2) to distinguish between words. But `keep_blank_chars=True` tells the method to only use (2).
